packages/suite/src/components/suite/modals/AddAccount/index.tsx

```diff
@@ -33,8 +34,8 @@ interface Props {
 }
 
 export const AddAccount = ({ device, onCancel, symbol, noRedirect }: Props) => {
-    const { changeAccountVisibility, changeCoinVisibility, goto } = useActions({
-        changeAccountVisibility: accountActions.changeAccountVisibility,
+    const dispatch = useDispatch();
```

Comment:
This idea is very hard to achieve and gives us zero benefits:
1. If you use `typeof` to dispatch returned from RTK configure store, you will get type `Dispatch<AnyAction>` anyway = zero benefit
2. If you want to construct this type manually like we do in Suite, you need to construct discriminated union from all actions and that means importing all actions into one package. But then you have your Action type, but you can't import it anywhere because it will create cyclic dependency. Could be hacked somehow, but definitely not good.
3. Even if you achieve to create typed dispatch and use it everywhere, only benefit is that you have dispatch where you allow to dispatch only some actions. But do you need that? It is not going to protect you from any mistakes, it only limits dispatch to allowed actions, but in general all actions are allowed and when using action creators they always be valid = zero benefit.
4. Creating unions for Actions type is considered as antipattern https://phryneas.de/redux-typescript-no-discriminating-union

And why it's app dispatch mentioned in RTK docs? It has some benefits in two case. If you dispatching action object without action creators aka `dispatch({type: '...', payload});` which we will never do in suite-common/native or any refactored code. Second case is when you have truly modular application where you need to construct lot of different version of app with different modules (which we don't have and probably never will). And this is also not necessary because benefits of typed dispatch even in modular application is so small that we didn't do that on my previous project anyway and it was really super modular app.

Reply:
I was mostly thinking about including the `ThunkDispatch` [type](https://redux-toolkit.js.org/tutorials/typescript#define-typed-hooks). The Action type union is an overkill, in my opinion.

Reply:
I see, I will do that in separate PR, probably little bit differently.